tools/print_tock_memory_usage.py

```diff
@@ -0,0 +1,422 @@
+#!/usr/bin/env python3
+
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#
+# Prints out the memory usage of a Tock kernel binary ELF.
+# Currently only works on ARM binaries.
+#
+# Usage: print_tock_memory_usage.py ELF
+#
+# Author: Philip Levis <philip.levis@gmail.com>
+
+# pylint: disable=superfluous-parens
+'''
+Script to print out the memory usage of a Tock kernel binary ELF.
+
+Usage: print_memory_usage.py ELF
+Options:
+  -dn, --depth=n      Group symbols at depth n or greater. E.g.,
+                      depth=2 will group all h1b::uart:: symbols
+                      together. Default: 1
+  -v, --verbose       Print verbose output.
+  -s, --show-waste    Show where RAM is wasted (due to padding)
+'''
+
+import os
+import re
+import sys
+import getopt
+import cxxfilt   # Demanging C++/Rust symbol names
```

Comment:
Why change this file in this PR?

Reply:
No idea -- shouldn't be.

Reply:
It's included in the PR. It looks like you need to rebase on master as you have ~60 commits here.